scripts/dockerfiles/run.sh

```diff
@@ -78,9 +78,35 @@ service rsyslog start
 cron
 
 # Run custom post config role scripts
+# Run any custom scripts other than pre-start
 {{ range $script := .role.PostConfigScripts}}
+{{ if not (is_pre_start $script) }}
+    echo bash {{ if not (is_abs $script) }}/opt/hcf/startup/{{ end }}{{ $script }}
     bash {{ if not (is_abs $script) }}/opt/hcf/startup/{{ end }}{{ $script }}
 {{ end }}
+{{ end }}
+
+# Run all the scripts called pre-start, but ensure consul_agent/bin/pre-start is run before others.
+# None of the other pre-start scripts appear to have any dependencies on one another.
+function sorted-pre-start-paths()
+{
+    declare -a fnames
+    idx=0
+    if [ -x /var/vcap/jobs/consul_agent/bin/pre-start ] ; then
+	fnames[$idx]=/var/vcap/jobs/consul_agent/bin/pre-start
+	idx=$((idx + 1))
+    fi
+    for fname in $(find /var/vcap/jobs/*/bin -name pre-start | grep -v '/consul_agent/bin/pre-start$') ; do
+	fnames[$idx]=$fname
+	idx=$((idx + 1))
+    done
+    echo ${fnames[*]}
```

Comment:
In general it makes more sense to invoke 'grep' once rather than test _n_ times, but in reality there are never more than 3 pre-start files in a given role, and usually just 1, so doing individual tests are ok. Keep in mind that `test` isn't a shell builtin.